### ISSUE TITLE
Encoding of Properties files - ISO-8859-1 vs UTF-8

### ISSUE THREAD

**Issue by MikeEdgar** [smallrye/smallrye-config#995]
Started in https://github.com/smallrye/smallrye-open-api/issues/1547

SmallRye Config loads properties using a UTF-8 decoding InputStreamReader [1]. However, the Properties JavaDoc [2] indicates that loading an InputStream is decoded as ISO-8859-1 (Latin-1). The result is that users with valid Latin-1 characters in their properties files may end up with corrupted entries from the config source.

[1] https://github.com/smallrye/smallrye-config/blob/170642b18ee47c400e00b4b5afe2d0b8f7d2be85/common/src/main/java/io/smallrye/config/common/utils/ConfigSourceUtil.java#L77-L87
[2] https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Properties.html

**Comment by dmlloyd** [comment]
This was intentional. Properties files used for configuration may commonly contain characters not directly representable in Latin-1 (particularly when used in countries outside of North America and western Europe, or when encoding other Unicode characters such as emoji or symbols). If Latin-1 encoding were used in this case, these files would not be easily human-readable and almost impossible for humans to author, which conflicts with the intent of the project. Therefore the choice was made to require the much more ubiquitous UTF-8 encoding for properties files read by SmallRye Config by default. The `Reader`-based API we used on `java.util.Properties` was added in Java 6 primarily for the purpose of allowing other, more modern encodings in application-specific properties files.

Properties files using other encodings must be transcoded into UTF-8 format, either within your IDE or using a command-line tool such as `iconv`, in order to be usable by SmallRye Config.
